### PR TITLE
feat: Added is_guild_owner for dpp::guild_member

### PR DESCRIPTION
**Pull request by AA1999**
…member is the guild owner.

Added a helper method so that checking if the member is the guild owner is easier.

## Code change checklist

- [✅] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [✅] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [✅] I tested that my change works before raising the PR.
- [✅] I have ensured that I did not break any existing API calls.
- [✅] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

File: src/dpp/guild.cpp [hunk 1]
```diff
@@ -347,6 +347,11 @@ bool guild_member::has_rejoined() const {
 	return flags & dpp::gm_did_rejoin;
 }
 
+bool guild_member::is_guild_owner() const {
+	auto* _guild = find_guild(user_id);
+	return _guild != nullptr && _guild->owner_id == this->user_id;
```
Comment (Mishura4) [inline src/dpp/guild.cpp:352]: So if the guild is not found in the cache this returns false, this is a bit weird, do we have any other functions in D++ that do this or is this potentially going to surprise people? @braindigitalis

Comment (braindigitalis) [reply src/dpp/guild.cpp:352]: it might surprise people who have cache off yes

Comment (braindigitalis) [reply src/dpp/guild.cpp:352]: it should probably be a method of dpp::guild that takes a user. not really needed though you can just check user id == guild owner id where you have both.

most people do have guild cache on. best to clearly document this assumption in the docs of the function.

File: src/dpp/guild.cpp [hunk 2]
```diff
@@ -347,6 +347,11 @@ bool guild_member::has_rejoined() const {
 	return flags & dpp::gm_did_rejoin;
 }
 
+bool guild_member::is_guild_owner() const {
+	auto* _guild = find_guild(user_id);
```
Comment (braindigitalis) [inline src/dpp/guild.cpp:351]: this is wrong. you're using user id to search for a guild

**Review by braindigitalis — CHANGES_REQUESTED** [review]

File: src/dpp/guild.cpp (hunk 2 above)
Comment (AA1999) [reply src/dpp/guild.cpp:351]: Yeah I might have messed it up, will fix

File: include/dpp/guild.h [hunk 3]
```diff
@@ -520,6 +520,14 @@ class DPP_EXPORT guild_member : public json_interface<guild_member> {
 	 */
 	bool has_rejoined() const;
 
+	/**
+	 * @brief Returns true if the user is the guild owner.
+	 * @return true if the user is the guild owner.
+	 * @return false if the user is not the guild owner or the guild is not in the cache.
+	 * @note If dpp cache is disabled (which should be enabled by default) this function will always return false.
```
Comment (braindigitalis) [inline include/dpp/guild.h:527]: this should be reworded:
```cpp
/*
 * @note If the guild cache is disabled in the cluster constructor, this function will always return false.
 */
```
we dont need to tell them that its on by default, other docs indicate this

**Review by Jaskowicz1 — APPROVED** [review]

**Comment by Jaskowicz1** [comment]
Renamed this PR to correctly align with our conventions.

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**